### PR TITLE
[WIP] Add verbose flag - fixes #763

### LINKED ISSUE #763: yarn install --verbose
**Do you want to request a _feature_ or report a _bug_?**

Feature : This would make diagnosing installation issues much easier. See #764 for more context.

**What is the current behavior?**

Running `yarn --verbose` or `yarn install --verbose` does not display extended information about the installation process.

**What is the expected behavior?**

Should provide extended output similar to `npm i --verbose`

**Please mention your node.js, yarn and operating system version.**
- yarn@v0.15.1
- node@v4.4.6
- osx@10.10.5

### PR DESCRIPTION
**Pull request by kittens**
Work in progress, need to add more places to log. Adds verbose logging in the following places:

 - CLI errors.
 - File copying/extraneous etc.
 - HTTP requests. (Still needs more info)

### CODE REVIEW

File: __tests__/commands/global.js [hunk 1]
```diff
@@ -22,31 +22,31 @@ async function runGlobal(
   name: string,
   checkGlobal?: ?(config: Config, reporter: Reporter, out: string) => ?Promise<void>,
 ): Promise<void> {
+  let out = '';
+  const stdout = new stream.Writable({
```
Comment (Daniel15) [inline __tests__/commands/global.js]: This is copied and pasted in a few places, what about making a `mockOutputStream` module to encapsulate it?

Comment (kittens) [inline __tests__/commands/global.js]: Yeah i noticed this too, I'm going to abstract it out in a subsequent pull request.

**Comment by bestander** [comment]
ping @kittens, a few lint and undefined errors

**Comment by bestander** [comment]
Looks great!
Merge when ready

File: __tests__/commands/_helpers.js [hunk 2]
```diff
@@ -68,32 +68,31 @@ export async function run<T, R>(
...
-    write(data, encoding, cb) {
-      out += data;
-      cb();
-    },
-  });
-
-  const reporter = new Reporter({stdout, stderr: stdout});
-
   if (beforeInstall) {
     await beforeInstall(cwd);
```
Comment (onemen) [inline __tests__/commands/_helpers.js]: Can you add reporter and out arguments to before install?

```
  if (beforeInstall) {
-   await beforeInstall(cwd);
+   await beforeInstall(cwd, reporter, () => out);
  }

  // remove the lockfile if we create one and it didn't exist before
  const lockfile = await createLockfile(cwd);

  // create directories
  await fs.mkdirp(path.join(cwd, '.yarn-global'));
```

**Comment by Aarbel** [comment]
@bestander yarn --verbose isn't mentioned in the documentation :
https://yarnpkg.com/lang/en/docs/cli/install/

could you add it ? ;)

**Comment by bestander** [comment]
Please send a PR, we really appreciate when community helps us with
improving the documentation.

**Comment by Aarbel** [comment]
ok

**Pull request merged**